crates/bin/prove_block/src/main.rs

```diff
@@ -478,40 +511,74 @@ async fn main() -> Result<(), Box<dyn Error>> {
         .visited_pcs
         .iter()
         .map(|(class_hash, visited_pcs)| {
-            let class_hash_felt = felt_api2vm(class_hash.0);
-            let compiled_class_hash_felt = class_hash_to_compiled_class_hash.get(&class_hash_felt).unwrap();
-            (*compiled_class_hash_felt, visited_pcs.iter().copied().map(Felt252::from).collect::<Vec<_>>())
+            (felt_api2vm(class_hash.0), visited_pcs.iter().copied().map(Felt252::from).collect::<Vec<_>>())
         })
         .collect();
 
+    log::debug!("class_hash_to_compiled_class_hash ({}):", class_hash_to_compiled_class_hash.len());
+    for (ch, cch) in &class_hash_to_compiled_class_hash {
+        log::debug!("    0x{:x} => 0x{:x}", ch, cch);
+    }
+
+    // TODO: ugly clone here
+    // Convert the Blockifier storage into an OS-compatible one
+    // let (contract_storage_map, previous_state, updated_state) =
+    //     reexecute::build_starknet_storage_async(blockifier_state,
+    // initial_state.clone()).await.unwrap();
+
+    // TODO: how should special contracts (e.g. 1) be handled?
+    // contract_storage_map.insert(
+    // Felt252::ONE,
+    // OsSingleStarknetStorage::new(
+    // PatriciaTree::empty_tree(&mut initial_state.ffc, Height(251), SimpleLeafFact::empty()).await?,
+    // PatriciaTree::empty_tree(&mut initial_state.ffc, Height(251), SimpleLeafFact::empty()).await?,
+    // &[],
+    // initial_state.ffc.clone()).await?,
+    // );
+
+    // Pass all contract addresses as expected accessed indices
+    // let contract_indices: HashSet<TreeIndex> =
+    //     contract_states.keys().chain(contract_storages.keys()).map(|address|
+    // address.to_biguint()).collect(); let contract_indices: Vec<TreeIndex> =
+    // contract_indices.into_iter().collect();
+
+    // let contract_state_commitment_info =
+    //     CommitmentInfo::create_from_expected_updated_tree::<_, PedersenHash, ContractState>(
+    //         previous_state.contract_states.clone(),
+    //         updated_state.contract_states.clone(),
+    //         &contract_indices,
+    //         &mut initial_state.ffc,
+    //     )
+    //     .await
+    //     .unwrap_or_else(|e| panic!("Could not create contract state commitment info: {:?}", e));
+
+    // let contract_class_commitment_info =
+    //     CommitmentInfo::create_from_expected_updated_tree::<_, PoseidonHash, ContractClassLeaf>(
+    //         previous_state.contract_classes.clone().expect("previous state should have class trie"),
+    //         updated_state.contract_classes.clone().expect("updated state should have class trie"),
+    //         &contract_indices,
+    //         &mut initial_state.ffc.clone_with_different_hash::<PoseidonHash>(),
+    //     )
+    //     .await
+    //     .unwrap_or_else(|e| panic!("Could not create contract class commitment info: {:?}", e));
+
     let os_input = StarknetOsInput {
-        contract_state_commitment_info,
+        contract_state_commitment_info: Default::default(),
```

Comment:
Why are you putting a default?

Reply:
This isn't relevant until the end of the OS when we compute the state roots, which is coming $soon

Reply:
Also, see all the commented-out code above this -- it'll be implemented soon